Assets/Scripts/SS3D/Systems/Health/HealthController.cs

```diff
@@ -23,17 +21,28 @@ private void BecomeGhost(GameObject player, GameObject ghost)
         Mind originMind = originEntity.Mind;
 
         ghostEntity.SetMind(originMind);
-        destroyObjects(originEntity, ghostEntity);
+        RpcDestroyObjects(originEntity);
+        RpcUpdateGhostPosition(originEntity, ghostEntity);
+    }
+
+    [ObserversRpc]
+    private void RpcDestroyObjects(Entity originEntity)
+    {
+        GameObject originEntityGameObject = originEntity.gameObject;
+        originEntityGameObject.GetComponent<Hands>().Dispose(true);
+        originEntityGameObject.GetComponent<Inventory>().Dispose(true);
+        originEntityGameObject.GetComponent<InteractionController>().Dispose(true);
+        originEntityGameObject.GetComponent<StaminaController>().Dispose(true);
+        originEntityGameObject.GetComponent<HumanoidController>().Dispose(true);
+        
     }
 
     [ObserversRpc]
-    private void destroyObjects(Entity originEntity, Entity ghostEntity)
+    private void RpcUpdateGhostPosition(Entity originEntity, Entity ghostEntity)
     {
-        originEntity.gameObject.GetComponent<StaminaController>().Dispose(true);
-        originEntity.gameObject.GetComponent<HumanoidController>().Dispose(true);
-        originEntity.gameObject.transform.Rotate(new Vector3(90, 0, 0));
-        ghostEntity.transform.position = originEntity.transform.position;
-        ghostEntity.transform.rotation = originEntity.transform.rotation;
+        Transform originEntityTransform = originEntity.transform;
```

Comment:
You don't need to cache the transform, we already have that inside the NetworkActor class. You can have something like:

```
ghostEntity.Transform.SetPositionAndRotation(originEntityTransform.position, originEntityTransform.rotation);
originEntity.Transform.Rotate(new Vector3(90, 0, 0));
```